micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jvm/CompilationMetricsTest.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.jvm;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+class CompilationMetricsTest {
+    @Test
+    void compilationTimeMetric() {
+        MeterRegistry registry = new SimpleMeterRegistry();
+        new CompilationMetrics().bindTo(registry);
+
+        assertThat(registry.get("jvm.compilation.time.total").gauge().value()).isGreaterThan(0);
```

Comment:
Will this test fail on builds that run on alternate JDKs?  I'm not too familiar with the build setup in that regard @shakuzen

Reply:
I was of the false memory that we already had some kind of matrix build which don't just test the different Hotspot JVMs. Apparently that isn't the case. Was quickly testing a `./gradlew test` with 8u222-openJ9 and there's quite some test failures before I could reach this one. (Hope I didn't screw up.)

Reply:
Is it a problem in the end?

Reply:
> I was of the false memory that we already had some kind of matrix build which don't just test the different Hotspot JVMs. Apparently that isn't the case.

I opened an issue for it but never got around to doing it - #1462. If a test is specific to a JVM, you can use JUnit Assumptions to skip it when those assumptions aren't meant. We do this for example in the `ProcessorMetricsTest`: https://github.com/micrometer-metrics/micrometer/blob/49162a881c4c80976980d6ea0ef45d2d59c4ca14/micrometer-core/src/test/java/io/micrometer/core/instrument/binder/system/ProcessorMetricsTest.java#L53